docs/site/src/plugins/inject-code/injectLoader.js

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+const fs = require("fs");
+const path = require("path");
+
+const addCodeInject = function (source) {
+  let fileString = source;
+  const callback = this.async();
+
+  const options = this.getOptions();
+  const markdownFilename = path.basename(this.resourcePath);
+  const repoPath = path.join(__dirname, "../../../../..");
+
+  // Do not load and render markdown files without docusaurus header.
+  // These files are only used to be included in other files and should not generate their own web page
+  if (fileString.length >= 3 && fileString.substring(0, 3) !== "---") {
+    return callback && callback(null, "");
+  }
+
+  function addMarkdownIncludes(fileContent) {
+    let res = fileContent;
+    const matches = fileContent.match(/\{@\w+: .+\}/g);
+    if (matches) {
+      matches.forEach((match) => {
+        const replacer = new RegExp(match, "g");
+        if (match.startsWith("{@inject: ")) {
+          const injectFileFull = match.substring(10, match.length - 1);
+          const injectFile = injectFileFull.substring(
+            0,
+            injectFileFull.indexOf("#") > 0
+              ? injectFileFull.indexOf("#")
+              : injectFileFull.length,
+          );
+          let fileExt = injectFile.substring(injectFile.lastIndexOf(".") + 1);
+          let language = "";
+          const fullPath = path.join(repoPath, injectFile);
+
+          // Assuming rust manifests and locks never being used
+          switch (fileExt) {
+            case "toml":
+              language = "move";
+              break;
+            case "lock":
+              language = "move";
+              break;
+            case "sh":
+              language = "shell";
+              break;
+            case "mdx":
+              language = "markdown";
+              break;
+            case "tsx":
+              language = "ts";
+              break;
+            default:
+              language = fileExt;
+          }
+
+          if (fs.existsSync(fullPath)) {
+            let injectFileContent = fs.readFileSync(fullPath, "utf8");
+            const marker =
+              injectFileFull.indexOf("#") > 0
+                ? injectFileFull.substring(injectFileFull.indexOf("#"))
+                : null;
+            if (marker) {
+              const regexStr = `\\/\\/\\s?docs::${marker.trim()}\\b([\\s\\S]*)\\/\\/\\s*docs::\\/\\s?${marker.trim()}\\b`;
```

Comment:
This assumes that `//` is the comment syntax in the language you are editing, but this isn't true among the languages the plugin supports (e.g. `toml`, and `sh` use `#`, `markdown` uses HTML comments, I guess: `<!-- ... -->`?)

Reply:
I thought about this, but only anticipated us using for move or ts. Short sited, for sure.